core/controllers/base_test.py

```diff
@@ -87,6 +87,18 @@ def test_requests_for_invalid_paths(self):
         response = self.testapp.put('/gallery/extra', {}, expect_errors=True)
         self.assertEqual(response.status_int, 404)
 
+    def test_redirect_in_both_logged_in_and_logged_out_states(self):
+        "Test for a redirect in both logged in and logged out states on '/'."
+
+        # Logged out state
+        response = self.testapp.get('/')
+        self.assertEqual(response.status_int, 302)
+
+        # Login and assert that there is a redirect
+        self.login('user@example.com')
+        response = self.testapp.get('/')
+        self.assertEqual(response.status_int, 302)
+
```

Comment:
FYI, it's good practice while writing this sort of test to do self.logout() at the end (and, more generally, to return things to their original state).